### PR TITLE
Don't create a whole set of subdirs in the temporary space…

### PR DESCRIPTION
**Pull request by ocaisa**
…slows down lmod a lot

This is an improvement for `--minimal-toolchains`

### CODE REVIEW

File: easybuild/framework/easyconfig/tools.py [hunk 1]
```diff
@@ -333,7 +333,10 @@ def find_minimally_resolved_modules(easyconfigs, avail_modules, existing_modules
     modtool = modules_tool()
     # copy, we don't want to modify the origin list of available modules
     avail_modules = avail_modules[:]
-    minimal_ecs_dir = tempfile.mkdtemp(prefix='minimal-easyconfigs-')
+    # Create a (temporary sub-)directory to store minimal easyconfigs
+    minimal_ecs_dir = os.path.join(tempfile.tempdir,'minimal-easyconfigs')
+    if not os.path.exists(minimal_ecs_dir):
+        os.mkdir(minimal_ecs_dir)
```
Comment (boegel) [inline easybuild/framework/easyconfig/tools.py]: please use the `mkdir` function available from `filetools`:

```python
mkdir(minimal_ecs_dir, parents=True)
```

**Comment by boegel** [comment]
@ocaisa: have you tested this fix? the `tempfile.mkdtemp` should create a new subdirectory under the temporary directory that was created for this EB session specifically...

Are you seeing something else?

Also, why does this slow down Lmod exactly, since we're talking easyconfigs here, not modules?

**Comment by ocaisa** [comment]
Ok, you are right about teh lmod I hadn't really thought it through. I was just running some test for the other PR and I noticed that in the temporary space things were unnecessarily complex:
```
alanc@alanc-VirtualBox:~/EasyBuild_Git/easybuild-framework$ eb --dry-run /home/alanc/EasyBuild_Git/easybuild-easyconfigs/easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.6.3-iccifort-2011.13.367.eb --experimental --minimal-toolchains --disable-cleanup-tmpdir --add-dummy-to-minimal-toolchains 
== temporary log file in case of crash /tmp/eb-sWUyL0/easybuild-3L65c3.log
Dry run: printing build status of easyconfigs and dependencies
 * [ ] /tmp/eb-sWUyL0/minimal-easyconfigs-Z5LTOV/icc-2011.13.367.eb (module: Core | icc/2011.13.367)
 * [ ] /tmp/eb-sWUyL0/minimal-easyconfigs-I8gA27/ifort-2011.13.367.eb (module: Core | ifort/2011.13.367)
 * [ ] /tmp/eb-sWUyL0/minimal-easyconfigs-v0xeXR/iccifort-2011.13.367.eb (module: Core | iccifort/2011.13.367)
 * [ ] /tmp/eb-sWUyL0/minimal-easyconfigs-cmp_yg/hwloc-1.6-iccifort-2011.13.367.eb (module: Compiler/intel/2011.13.367 | hwloc/1.6)
 * [ ] /tmp/eb-sWUyL0/minimal-easyconfigs-4Aarf7/OpenMPI-1.6.3-iccifort-2011.13.367.eb (module: Compiler/intel/2011.13.367 | OpenMPI/1.6.3)
== Keeping temporary log file(s) /tmp/eb-sWUyL0/easybuild-3L65c3.log* and directory /tmp/eb-sWUyL0.
alanc@alanc-VirtualBox:~/EasyBuild_Git/easybuild-framework$ ls /tmp/eb-sWUyL0
easybuild-3L65c3.log        minimal-easyconfigs-I8gA27  minimal-easyconfigs-sMxu19  minimal-easyconfigs-wJL2JV
minimal-easyconfigs-4Aarf7  minimal-easyconfigs-PAY4lc  minimal-easyconfigs-v0xeXR  minimal-easyconfigs-Z5LTOV
minimal-easyconfigs-cmp_yg  minimal-easyconfigs-RB0QPc  minimal-easyconfigs-WF78_c
```
With this fix, things become:
```
alanc@alanc-VirtualBox:~/EasyBuild_Git/easybuild-framework$ eb --dry-run /home/alanc/EasyBuild_Git/easybuild-easyconfigs/easybuild/easyconfigs/o/OpenMPI/OpenMPI-1.6.3-iccifort-2011.13.367.eb --experimental --minimal-toolchains --disable-cleanup-tmpdir --add-dummy-to-minimal-toolchains 
== temporary log file in case of crash /tmp/eb-6ORO3Y/easybuild-Jf6tkW.log
Dry run: printing build status of easyconfigs and dependencies
 * [ ] /tmp/eb-6ORO3Y/minimal-easyconfigs/icc-2011.13.367.eb (module: Core | icc/2011.13.367)
 * [ ] /tmp/eb-6ORO3Y/minimal-easyconfigs/ifort-2011.13.367.eb (module: Core | ifort/2011.13.367)
 * [ ] /tmp/eb-6ORO3Y/minimal-easyconfigs/iccifort-2011.13.367.eb (module: Core | iccifort/2011.13.367)
 * [ ] /tmp/eb-6ORO3Y/minimal-easyconfigs/hwloc-1.6-iccifort-2011.13.367.eb (module: Compiler/intel/2011.13.367 | hwloc/1.6)
 * [ ] /tmp/eb-6ORO3Y/minimal-easyconfigs/OpenMPI-1.6.3-iccifort-2011.13.367.eb (module: Compiler/intel/2011.13.367 | OpenMPI/1.6.3)
== Keeping temporary log file(s) /tmp/eb-6ORO3Y/easybuild-Jf6tkW.log* and directory /tmp/eb-6ORO3Y.
alanc@alanc-VirtualBox:~/EasyBuild_Git/easybuild-framework$ ls /tmp/eb-6ORO3Y
easybuild-Jf6tkW.log  minimal-easyconfigs
```

**Comment by boegel** [comment]
@ocaisa: OK, that's also a good reason to apply this patch. Please tackle the remark though.

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@ocaisa: let me know if you need help figuring out the broken test

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
@ocaisa: https://github.com/ocaisa/easybuild-framework/pull/25 fixes the broken test

**Comment by boegel** [comment]
Going in, thanks for the fix @ocaisa!

**Pull request merged**